include/bm/bm_sim/extern.h

```diff
@@ -61,20 +63,44 @@ class ExternFactoryMap {
            #extern_name,                                                \
            [](){ return std::unique_ptr<ExternType>(new extern_name()); });
 
-#define BM_REGISTER_EXTERN_METHOD(extern_name, extern_method_name, ...) \
-  template <typename... Args>                                           \
-  struct _##extern_name##_##extern_method_name##_0                      \
-      : public ::bm::ActionPrimitive<::bm::ExternType *, Args...> {     \
+#define BM_REGISTER_EXTERN_W_NAME(extern_name, extern__)                \
```

Comment:
Can we use `BM_REGISTER_EXTERN_W_NAME` to implement `BM_REGISTER_EXTERN` to avoid duplicating the code, or is there something preventing us from doing that? (same for method below)